OrbitVulkanLayer/QueueManager.cpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "QueueManager.h"
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_vulkan_layer {
+
+void QueueManager::TrackQueue(const VkQueue& queue, const VkDevice& device) {
+  absl::WriterMutexLock lock(&mutex_);
+  CHECK(!queue_to_device_.contains(queue) || queue_to_device_.at(queue) == device);
+  queue_to_device_[queue] = device;
+}
+
+const VkDevice& QueueManager::GetDeviceOfQueue(const VkQueue& queue) {
+  absl::ReaderMutexLock lock(&mutex_);
+  CHECK(queue_to_device_.contains(queue));
```

Comment:
Have you considered returning a std::optional<VkDevice> instead of having a CHECK?

Reply:
As for the other CHECK, I'd really like to use it to encode certain expectations on that call. The logic should never call this method, when the queue is not there, so I wanted to encode this here.